dts/arm/ti/cc13xx_cc26xx.dtsi

```diff
@@ -123,11 +123,12 @@
 			status = "disabled";
 		};
 
+		/* The RTC peripheral backs the kernel system clock and tick timer. */
 		rtc: rtc@40092000 {
-			compatible = "ti,cc13xx-cc26xx-rtc";
+			compatible = "ti,cc13xx-cc26xx-rtc-timer";
 			reg = <0x40092000 0x1000>;
 			interrupts = <4 0>;  /* interrupt #20 = 4 + 16 */
-			status = "disabled";
+			status = "okay"; /* the system clock timer is mandatory */
```

Comment:
Do we really need to default enable this at an SoC family level? I am not sure if this is required for a very minimal system if not I think it would be preferred to keep the existing scheme, so that users using different hardware platforms are not forced with defaults that they are not aware/needed for a minimal system.

Reply:
@vaishnavachath The system clock/timer driver is required in all cases that I'm aware of, as basic kernel thread scheduling depends on it. AFAIK several SoCs don't even bother to expose the system clock driver via DT and just mandatorily add it in the background.

In the rare case where s.o. actually wants to disable this (e.g. to provide an alternative sysclock driver), an overlay can still be created that disables the (sane) default. Someone with such an expert requirement will probably know how to do that. IMO it's much more probable, that less knowledgeable users will forget to add this peripheral for their custom boards which will provoke strange error messages they don't understand.

But maybe I overlook something? Let me know...